src/graphql/client.ts

```diff
@@ -1,18 +1,31 @@
 // DON'T TOUCH THIS
 // These are separate clients and do not share configs between themselves
 import { ApolloClient, InMemoryCache } from "@apollo/client";
+import { ApolloLink } from "@apollo/client/link/core";
 import { createFetch, createSaleorClient } from "@saleor/sdk";
 import { createUploadLink } from "apollo-upload-client";
 
 import { API_URI } from "../config";
 import introspectionQueryResultData from "./fragmentTypes.generated";
 import { TypedTypePolicies } from "./typePolicies.generated";
 
-export const link = createUploadLink({
-  credentials: "include",
-  uri: API_URI,
-  fetch: createFetch(),
-});
+const attachVariablesLink = new ApolloLink((operation, forward) =>
+  forward(operation).map(data => ({
+    ...data,
+    extensions: {
+      ...data.extensions,
+      variables: operation.variables,
+    },
+  })),
+);
+
+export const link = attachVariablesLink.concat(
+  createUploadLink({
+    credentials: "include",
+    uri: API_URI,
+    fetch: createFetch(),
+  }),
+);
```

Comment:
Why is this change needed?

Reply:
We need access not only to response but to variables too - since request count is not constant and queries don't have identifiers we need another way to map responses to changes in datagrid.